Snowflake.Data.Tests/IntegrationTests/ConnectionChangedSessionIT.cs

```diff
@@ -0,0 +1,160 @@
+using NUnit.Framework;
+using Snowflake.Data.Client;
+using Snowflake.Data.Core;
+using Snowflake.Data.Core.Session;
+using Snowflake.Data.Tests.Util;
+
+namespace Snowflake.Data.Tests.IntegrationTests
+{
+    [TestFixture]
+    [NonParallelizable]
+    public class ConnectionChangedSessionIT : SFBaseTest
+    {
+        private readonly QueryExecResponseData _queryExecResponseChangedRole = new()
```

Comment:
BTW do we want it to work for old pool as well? As I see the implementation - it would work for old pool as well. Don't we want to cover that in tests?

Reply:
lets have it only in latest pool (the least changes to prev the better)